### PR TITLE
[WIP] Add Lele's part of the training about logs and status

### PR DESCRIPTION
**Pull request by eldios**
Working on adding my part of the training about logs and status messages.

Also fixed a couple stray line and spaces.

### CODE REVIEW

File: gravity_upgrade.md [hunk 1]
```diff
@@ -72,25 +73,126 @@ A manual upgrade operation starts with an operation plan which is a tree of acti
...
+    It's particularly useful to verify at which step if an operation gets stuck,
+    and thus analyze the root cause and to resume once fixed.
+  + content of `gravity-system.log` and (if present) `gravity-install.log` files.
+    These files are usually found in the installation directory
+    or in newer versions they will be located in /var/log/gravity.
+    These files store the output that Gravity generates through installation and
+    it's mostly useful during troubleshooting.
+  + `sudo ./gravity status` - from the installer directory
+    The `status` command output shows the current state of Gravity clusters with
+    any hints about possible causes and hints of a degraded state.
```
Comment (aelkugia) [inline gravity_upgrade.md:94]: I would rephrase the second part to: ...with a high level overview of a cluster's health, hints if the cluster is in a degraded state, and the status of individual components.

File: gravity_upgrade.md [hunk 2]
```diff
@@ -72,25 +73,126 @@ A manual upgrade operation starts with an operation plan which is a tree of acti
...
+  
+### DURING RUNTIME
+  + `sudo gravity status`
+    The `status` command output shows the current state of Gravity clusters with
+    any hints about possible causes and hints of a degraded state.
```
Comment (aelkugia) [inline gravity_upgrade.md:99]: same as above with rephrasing

File: gravity_upgrade.md [hunk 3]
```diff
@@ -72,25 +73,126 @@ A manual upgrade operation starts with an operation plan which is a tree of acti
...
+  + `sudo gravity exec planet status`
+    The `planet status` command output shows the current state of Gravity
+    components and it's helpful to pin down which specific component is failing
+    on which node. It's a bit more verbose that `gravity status` but usually
```
Comment (aelkugia) [inline gravity_upgrade.md:103]: that -> than

File: gravity_upgrade.md [hunk 4]
```diff
@@ -72,25 +73,126 @@ A manual upgrade operation starts with an operation plan which is a tree of acti
...
+    more informative.
+  + `sudo gravity plan`
+    Once completed, this command analyze last operation ran on the Gravity cluster.
```
Comment (aelkugia) [inline gravity_upgrade.md:106]: analyze -> analyzes the

File: gravity_upgrade.md [hunk 5]
```diff
@@ -72,25 +73,126 @@ A manual upgrade operation starts with an operation plan which is a tree of acti
...
+
+  + OUTSIDE PLANET:
+    - `sudo systemctl list-unit-files '*grav*'`
+      This command will help you identify Gravity systemd units. Usually there
+      should only one `planet` and one `teleport` unit.
```
Comment (aelkugia) [inline gravity_upgrade.md:111]: only -> only be

File: gravity_upgrade.md [hunk 6]
```diff
@@ -72,25 +73,126 @@ A manual upgrade operation starts with an operation plan which is a tree of acti
...
+    - `sudo journalctl -f -u 'registry'`
+      This command will show all logs from the container registry, which is used to
+      distribute containers' data among all nodes.
+    - `sudo journalctl -f -u 'planet-agent'`
+      This command will show all logs from the planet-agent daemon, which is used to
+      exchange messages in between the different nodes in Gravity clusters.
+
+### Collect logfiles for reporting purpose
+  + `sudo gravity report reportfile.tgz`
+  + content of `gravity-system.log` and (if present) `gravity-install.log` files.
```
Comment (aelkugia) [inline gravity_upgrade.md:150]: Not just these files right? report will collect full cluster diagnostics into an archive

File: gravity_upgrade.md [hunk 7]
```diff
@@ -72,25 +73,126 @@ A manual upgrade operation starts with an operation plan which is a tree of acti
...
+    These files are usually found in the installation directory
+    or in newer versions they will be located in /var/log/gravity
+
```
Comment (aelkugia) [inline gravity_upgrade.md:153]: In `5.5.49` we added `---since` flag so I would add that example, customers routinely give feedback about the size of report. Also several new system collectors added, related PR [#1719](https://github.com/gravitational/gravity/pull/1719)

File: gravity_upgrade.md [hunk 8]
```diff
@@ -72,25 +73,126 @@ A manual upgrade operation starts with an operation plan which is a tree of acti
...
+### Nifty features coming in future versions
+Versions of Gravity from 6.1.x onward also includes a nice feature showing the
```
Comment (aelkugia) [inline gravity_upgrade.md:155]: `gravity status history` I believe is in [5.5.45](https://gravitational.com/gravity/docs/changelog/#5545-lts-may-18th-2020) as well

File: gravity_upgrade.md [hunk 9]
```diff
@@ -72,25 +73,126 @@ A manual upgrade operation starts with an operation plan which is a tree of acti
...
+2020-07-30T16:30:57Z [Node Degraded]    node=172_28_128_3.pensivevillani4966
+2020-07-30T16:31:07Z [Probe Failed]     node=172_28_128_3.pensivevillani4966    checker=etcd-healthz
+2020-07-30T16:32:05Z [Probe Succeeded]  node=172_28_128_3.pensivevillani4966    checker=etcd-healthz
+2020-07-30T16:32:05Z [Node Healthy]     node=172_28_128_3.pensivevillani4966
+2020-07-30T16:32:46Z [Probe Failed]     node=172_28_128_3.pensivevillani4966    checker=node-status
+2020-07-30T16:32:46Z [Probe Failed]     node=172_28_128_3.pensivevillani4966    checker=etcd-healthz
+2020-07-30T16:33:07Z [Probe Succeeded]  node=172_28_128_3.pensivevillani4966    checker=node-status
+2020-07-30T16:34:05Z [Probe Succeeded]  node=172_28_128_3.pensivevillani4966    checker=etcd-healthz
+2020-07-30T16:34:05Z [Node Healthy]     node=172_28_128_3.pensivevillani4966
+```
```
Comment (aelkugia) [inline gravity_upgrade.md:183]: I would add a few other things from `5.5.49` and `5.5.50` - that pertain to customer `M`

- **5.5.49:** `gravity status` - now display unhealhty critical system pods ([#1707](https://github.com/gravitational/gravity/pull/1707)) + added visibility into implicit controller status loop if cluster is in degraded state ([#1707](https://github.com/gravitational/gravity/pull/1707))
- **5.5.49:** Should we mention anything about pre-upgrade checks? If so [#1731](https://github.com/gravitational/gravity/pull/1731) added check to see if previous upgrade operation has succeeded/if failed plan was rolled back.
- **5.5.50:** mention `gravity plan resume` launching in background job to execute phases by default, adding `--block` resume plan in foreground, and the new command `gravity plan watch` to track the plan progress ([#1807](https://github.com/gravitational/gravity/issues/1807))
- **5.5.50:** ability to view/collect gravity cli command history `journalctl -t gravity-cli` ([#1799](https://github.com/gravitational/gravity/issues/1799))
- **5.5.50:** pre-upgrade check to see if teleport nodes are unavailable ([#1817](https://github.com/gravitational/gravity/issues/1817))

File: gravity_upgrade.md [hunk 10]
```diff
@@ -72,25 +73,126 @@ A manual upgrade operation starts with an operation plan which is a tree of acti
...
+The list is divided based on when the specific command will be more likely used, like during upgrade or at runtime.
+
+### DURING INSTALL / UPGRADE
+  + `sudo ./gravity plan` - from the installer directory
+    This command is used to analyze the progress through any Gravity operations.
+    It's particularly useful to verify at which step if an operation gets stuck,
+    and thus analyze the root cause and to resume once fixed.
+  + content of `gravity-system.log` and (if present) `gravity-install.log` files.
+    These files are usually found in the installation directory
+    or in newer versions they will be located in /var/log/gravity.
```
Comment (r0mant) [inline gravity_upgrade.md:89]: Just /var/log. So e.g. `/var/log/gravity-system.log`. Also, I don't think upgrade creates gravity-install.log.

File: gravity_upgrade.md [hunk 11]
```diff
@@ -72,25 +73,126 @@ A manual upgrade operation starts with an operation plan which is a tree of acti
 
 ## Gravity Upgrade Status, Logs, and more
 
+In order to monitor the cluster, it's useful to go through logs and status messages provided by the platform.
+
+Below there's a list of useful commands/resources to use during debugging.
+
+The list is divided based on when the specific command will be more likely used, like during upgrade or at runtime.
+
+### DURING INSTALL / UPGRADE
```
Comment (r0mant) [inline gravity_upgrade.md:82]: Also need to add commands related to upgrade agents. How to check the agent is running (gravity-agent.service), how to check its logs, how to redeploy, etc. Plus, all of those things that @aelkugia mentioned below (resuming plan, checking CLI history, etc.).

File: gravity_upgrade.md [hunk 12]
```diff
@@ -72,25 +73,126 @@ A manual upgrade operation starts with an operation plan which is a tree of acti
...
+      distribute containers' data among all nodes.
+    - `sudo journalctl -f -u 'planet-agent'`
+      This command will show all logs from the planet-agent daemon, which is used to
+      exchange messages in between the different nodes in Gravity clusters.
+
+### Collect logfiles for reporting purpose
+  + `sudo gravity report reportfile.tgz`
+  + content of `gravity-system.log` and (if present) `gravity-install.log` files.
+    These files are usually found in the installation directory
+    or in newer versions they will be located in /var/log/gravity
```
Comment (r0mant) [inline gravity_upgrade.md:152]: We do not use /var/log/gravity directory for anything.

File: gravity_upgrade.md [hunk 13]
```diff
@@ -72,25 +73,126 @@ A manual upgrade operation starts with an operation plan which is a tree of acti
...
+      This command will show all logs from Docker, which is the software used to
+      run the actual containers used by Kubernetes to run its own Pods.
+    - `sudo journalctl -f -u 'registry'`
+      This command will show all logs from the container registry, which is used to
+      distribute containers' data among all nodes.
+    - `sudo journalctl -f -u 'planet-agent'`
+      This command will show all logs from the planet-agent daemon, which is used to
+      exchange messages in between the different nodes in Gravity clusters.
+
+### Collect logfiles for reporting purpose
```
Comment (r0mant) [inline gravity_upgrade.md:148]: This should also show how to collect a report from an individual node in case gravity-site is down (using lower-level `gravity system report` command).

File: gravity_upgrade.md (hunk 6 above)
Comment (eldios) [reply gravity_upgrade.md:150]: those are two different points... I'll try to split them so that it's more evident.

File: gravity_upgrade.md (hunk 7 above)
Comment (eldios) [reply gravity_upgrade.md:153]: As specified above, the `report` command was a different point... this was just to mention to analyze the `gravity-system.log` file as a separate source of info.

File: gravity_upgrade.md (hunk 9 above)
Comment (eldios) [reply gravity_upgrade.md:195]: I think most of this have now been added to the new commit. Closing this for now.

File: gravity_upgrade.md [hunk 14]
```diff
@@ -72,25 +73,154 @@ A manual upgrade operation starts with an operation plan which is a tree of acti
...
+  visibility into the controller status loop if the cluster is in a degraded
+  state
+* a new set of pre-upgrade checks were added to verify if previous upgrade
+  operation succeeded or a failed plan was rolled back
 
-## Exploring a Gravity Manual Upgrade
+* another set of pre-upgrade checks was added to verify if any Teleport node is
+  unavailable
 
+* `gravity plan resume` now launches its execution in background by default, but a new `--block` option was added to resume in foreground and a different command `gravity plan watch` was also introduced to track the plan progress
```
Comment (r0mant) [inline gravity_upgrade.md:208]: The command to follow the plan progress is actually `gravity plan --tail`, not watch.
